### PR TITLE
Feature/sim cards

### PR DESCRIPTION
**Pull request by rajjy1**

### CODE REVIEW

File: src/Telnyx.net/Services/Service.cs [hunk 1]
```diff
@@ -385,6 +386,32 @@ protected async Task<EntityReturned> UpdateEntityAsync(string id, BaseOptions op
...
+        /// BulkUpdateEntity.
+        /// </summary>
+        /// <param name="options">options</param>
+        /// <param name="requestOptions">requestOptions</param>
+        /// <param name="cancellationToken">cancellationToken</param>
+        /// <param name="postPath">postPath</param>
+        /// <returns>{EntityReturned}</returns>
+        protected Task<EntityReturned> BulkUpdateEntityAsync(BaseOptions options, RequestOptions requestOptions, CancellationToken cancellationToken, string postPath = null)
+        {
+            return this.PutRequestAsync<EntityReturned>(this.ClassUrl(), options, requestOptions, cancellationToken);
```
Comment (thedonmon) [inline src/Telnyx.net/Services/Service.cs:412]: need to add await here

Comment (rajjy1) [reply src/Telnyx.net/Services/Service.cs]: done.

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardsService.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+        public TelnyxCollection<SimCardRecord> Register(SimCardRegisterOptions createOptions, RequestOptions requestOptions = null)
+        {
+            return this.simCardRegisterService.Create(createOptions, requestOptions);
+        }
+
+        public Task<TelnyxCollection<SimCardRecord>> RegisterAsync(SimCardRegisterOptions createOptions, RequestOptions requestOptions = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return this.simCardRegisterService.CreateAsync(createOptions, requestOptions, cancellationToken);
```
Comment (thedonmon) [inline src/Telnyx.net/Services/Wireless/SimCards/SimCardsService.cs:99]: missing await keyword

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardEnableService.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+﻿namespace Telnyx.net.Services.Wireless.SimCards
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Telnyx.net.Entities.Wireless.SimCards;
+
+    /// <summary>
+    /// SimCardsEnableService
+    /// </summary>
+    public class SimCardEnableService : Service<SimCardRecord>,
```
Comment (thedonmon) [inline src/Telnyx.net/Services/Wireless/SimCards/SimCardEnableService.cs:10]: Make this class internal if not intended to be exposed

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardDisableService.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
+﻿namespace Telnyx.net.Services.Wireless.SimCards
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Telnyx.net.Entities.Wireless.SimCards;
+
+    /// <summary>
+    /// SimCardsDisableService
+    /// </summary>
+    public class SimCardDisableService : Service<SimCardRecord>,
```
Comment (thedonmon) [inline src/Telnyx.net/Services/Wireless/SimCards/SimCardDisableService.cs:10]: Make this class internal if not intended to be exposed

File: src/Telnyx.net/Entities/TelnyxCollection.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    using Newtonsoft.Json;
+    using Telnyx.Infrastructure;
+
+    /// <summary>
+    /// Telnyx List.
+    /// </summary>
+    /// <typeparam name="T">Object Type.</typeparam>
+
+    [JsonArray]
+    public class TelnyxCollection<T> : TelnyxEntity, IList<T>
```
Comment (thedonmon) [inline src/Telnyx.net/Entities/TelnyxCollection.cs:14]: Why create this new class and not TelynxList? There are a lot of methods on here that aren't implemented explicitly

File: src/Telnyx.net/Services/Service.cs [hunk 6]
```diff
@@ -359,6 +359,7 @@ protected async Task<EntityReturned> UpdateEntityAsync(string id, BaseOptions op
         {
             return await this.PatchRequestAsync<EntityReturned>(this.InstanceUrl(id), options, requestOptions, cancellationToken);
         }
+
```
Comment (thedonmon) [inline src/Telnyx.net/Services/Service.cs:362]: remove blank line

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardEnableService.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        INestedCreatableWithIdInMid<SimCardRecord, BaseOptions>
```
Comment (thedonmon) [inline src/Telnyx.net/Services/Wireless/SimCards/SimCardEnableService.cs:11]: Don't have to use this interface but also can override the PostPath property of Service.cs for the post path url

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardDisableService.cs [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        INestedCreatableWithIdInMid<SimCardRecord, BaseOptions>
```
Comment (thedonmon) [inline src/Telnyx.net/Services/Wireless/SimCards/SimCardDisableService.cs:11]: Same comment as the Enable Service

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardsService.cs (hunk 2 above)
Comment (rajjy1) [reply src/Telnyx.net/Services/Wireless/SimCards/SimCardsService.cs]: done

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardEnableService.cs (hunk 3 above)
Comment (rajjy1) [reply src/Telnyx.net/Services/Wireless/SimCards/SimCardEnableService.cs]: done

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardDisableService.cs (hunk 4 above)
Comment (rajjy1) [reply src/Telnyx.net/Services/Wireless/SimCards/SimCardDisableService.cs]: done

File: src/Telnyx.net/Entities/TelnyxCollection.cs (hunk 5 above)
Comment (rajjy1) [reply src/Telnyx.net/Entities/TelnyxCollection.cs:14]: TelnyxList doesn't implement ICollection. TelnyxCollection implements IList which in turn implements ICollection. So, they are serving different purpose. The methods that are not currently used are marked not implemented. Only used methods are implemented.

File: src/Telnyx.net/Services/Service.cs (hunk 6 above)
Comment (rajjy1) [reply src/Telnyx.net/Services/Service.cs:362]: done

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardDisableService.cs (hunk 8 above)
Comment (rajjy1) [reply src/Telnyx.net/Services/Wireless/SimCards/SimCardDisableService.cs]: done.

File: src/Telnyx.net/Services/Wireless/SimCards/SimCardEnableService.cs (hunk 7 above)
Comment (rajjy1) [reply src/Telnyx.net/Services/Wireless/SimCards/SimCardEnableService.cs]: i have addressed this.

**Pull request merged**